src/confdb/confdb.c

```diff
@@ -2438,3 +2456,146 @@ int confdb_certmap_to_sysdb(struct confdb_ctx *cdb,
+              CONFDB_MONITOR_ACTIVE_DOMAINS, "sssd",
+              ret, strerror(ret));
+        goto done;
+    }
+
+    ret = confdb_list_all_domain_names(tmp_ctx, cdb, &all_domains);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_OP_FAILURE,
+              "Failed retrieving all domain name "
+              "list, error [%d], description '%s'\n",
+              ret, strerror(ret));
+        goto done;
+    }
+
+    for (int idx = 0; all_domains[idx]; idx++) {
+        ret = confdb_get_domain_enabled(cdb, all_domains[idx], &enabled);
+        if (ret != EOK && ret != ENOENT) {
+            DEBUG(SSSDBG_OP_FAILURE,
+                  "Failed retrieving 'enabled' attribute from '%s' domain; "
+                  "error [%d], description '%s'\n",
+                  all_domains[idx],
+                  ret, strerror(ret));
+            goto done;
+        }
+
+        if (ret == ENOENT) continue;
+
+        if (enabled == 1 && ! string_in_list(all_domains[idx], domlist, false)) {
+            ret = add_string_to_list(tmp_ctx, all_domains[idx], &domlist);
+            if (ret != EOK) {
+                DEBUG(SSSDBG_OP_FAILURE,
+                    "Failed adding '%s' domain to domain list; "
+                    "error [%d], description '%s'\n",
+                    all_domains[idx],
+                    ret, strerror(ret));
+                goto done;
+            }
+        }
+
+        if (enabled == 0 && string_in_list(all_domains[idx], domlist, false)) {
+            ret = del_string_from_list(all_domains[idx], &domlist, false);
+            if (ret != EOK) {
+                DEBUG(SSSDBG_OP_FAILURE,
+                    "Failed deleting '%s' domain from domain list; "
+                    "error [%d], description '%s'\n",
+                    all_domains[idx],
+                    ret, strerror(ret));
+                goto done;
+            }
+        }
+    }
+
+    ret = EOK;
+    talloc_steal(ctx, domlist);
+    *result = domlist;
+
+done:
+    talloc_free(tmp_ctx);
+    return ret;
+}
+
+
+/**
+ * Retrieve the enabled attribute for a specific domain.
+ * @param cdb The database configuration context.
+ * @param domain The domain name.
+ * @param enabled The output variable; it can not be NULL; if the
+ * domain is explicitely enabled, *enabled is equal to 1; if the
+ * domain is explicitely disabled, *enabled is equal to 0; any
+ * other case, *enabled is -1.
+ * @param Return EOK is the operation happened properly and *enabled
+ * contain the value of the attribute; if no entry found for enabled
+ * attribute it returns NOENT, else an error code.
+ */
+static int confdb_get_domain_enabled(struct confdb_ctx *cdb, char* domain, int *enabled)
+{
+    TALLOC_CTX* tmp_ctx = NULL;
+    char* section = NULL;
+    char **values = NULL;
+    int ret = EOK;
+
+    section = talloc_asprintf(tmp_ctx, CONFDB_DOMAIN_PATH_TMPL, domain);
+    values = NULL;
+    ret = confdb_get_param(cdb, tmp_ctx, section, CONFDB_DOMAIN_ENABLED, &values);
```

Comment:
Use confdb_get_bool instead.

Reply:
Sorry but I can not use `confdb_get_bool` because I loose the state of when the enabled attribute is not used for a domain, and I need that information to modify the list properly. `confdb_get_bool` does not return **ENOENT** when the attribute does not exist, instead of it store the default value provided and return **EOK**.